src/opera/pge/dswx_s1/dswx_s1_pge.py

```diff
@@ -112,28 +110,87 @@ class DSWxS1PostProcessorMixin(PostProcessorMixin):
 
     In addition to the base functionality inherited from PostProcessorMixin, this
     mixin adds an output validation step to ensure that the output file(s) defined
-    by the RunConfig exist and are valid (TODO).
+    by the RunConfig exist and are valid.
 
     """
 
     _post_mixin_name = "DSWxS1PostProcessorMixin"
     _cached_core_filename = None
 
+    def _validate_output(self):
+        """
+        Evaluates the output file(s) generated from SAS execution to ensure:
+            - That the file(s) contains some content (size is greater than 0).
+            - That the .tif output files (burst data) end with 'B01_WTR',
+              'B02_BWTR', or 'B03_CONF'
+            - That the there are the same number of each type of file, implying
+              3 output bands per tile
+
+        """
+        burst_dict = {}
+        num_bursts = []
+        output_extension = '.tif'
+
+        # get all .tiff files
+        output_products = list(
+            filter(
+                lambda filename: output_extension in filename,
+                self.runconfig.get_output_product_filenames()
+            )
+        )
+
+        if not output_products:
+            error_msg = (f"No SAS output file(s) with '{output_extension}' extension "
+                         f"found within '{self.runconfig.output_product_path}'")
+
+            self.logger.critical(self.name, ErrorCode.OUTPUT_NOT_FOUND, error_msg)
+
+        for out_product in output_products:
+            if not getsize(out_product):
+                error_msg = f"SAS output file {out_product} was created, but is empty"
+
+                self.logger.critical(self.name, ErrorCode.INVALID_OUTPUT, error_msg)
+
+            #  Gather the burst output files into a dictionary
+            #     key = burst type (e.g. B01_WTR.tif)
+            #     value = list of filenames of this type (e.g. ['OPERA_L3_DSWx-S1_..._v0.1_B01_WTR.tif', ...]
+            if output_extension in out_product:
```

Comment:
Is this if statement necessary? You've already filtered the list of `out_products` to those that contain `output_extension` in the file name?